Index: src/js/rishson/enterprise/Globals.js
===================================================================
--- src/js/rishson/enterprise/Globals.js	(revision 47ecd34cb2dae39011106ea5af7e9576b670ef7e)
+++ src/js/rishson/enterprise/Globals.js	(revision )
@@ -20,6 +20,15 @@
      */
     TOPIC_NAMESPACE : '/rishson/enterprise/widget',
 
+    /**
+     * @field
+     * @name rishson.enterprise.Globals.SEND_REQUEST
+     * @type {String}
+     * @private
+     * @description This topic should be called if you want to send a request to the server
+     */
+     SEND_REQUEST : this._topicNamespace + '/send/request',
+
     constructor : function () {
         this.TOPIC_USER_LOGOUT = this.TOPIC_NAMESPACE + '/user/logout'
     }
Index: scripts/setup.sh
===================================================================
--- scripts/setup.sh	(revision 47ecd34cb2dae39011106ea5af7e9576b670ef7e)
+++ scripts/setup.sh	(revision )
@@ -9,6 +9,9 @@
 DOJODIR="dojo-release-${VERSION}-src"
 OUTDIR=$(cd "$OUTDIR" &> /dev/null && pwd || echo "")
 
+# create the dojo dir under src/js as this is excluded from the skeleton directory structure by a gitignor
+mkdir $OUTDIR
+
 if which wget >/dev/null; then
 	GET="wget --no-check-certificate -O -"
 elif which curl >/dev/null; then
Index: src/js/rishson/enterprise/control/Controller.js
===================================================================
--- src/js/rishson/enterprise/control/Controller.js	(revision 47ecd34cb2dae39011106ea5af7e9576b670ef7e)
+++ src/js/rishson/enterprise/control/Controller.js	(revision )
@@ -30,21 +30,12 @@
 
     /**
      * @field
-     * @name rishson.enterprise.widget._Widget._topicNamespace
-     * @type {String}
-     * @private
-     * @description This namespace is prepended to every topic
-     */
-    _topicNamespace : Globals.TOPIC_NAMESPACE,
-
-    /**
-     * @field
      * @name rishson.enterprise.widget._Widget.subList
      * @type {Object}
      * @description Object that contains the list of topics that any derived widget can listen out for
      */
     //@todo make this private with get/set so that contents can only be added to
-    subList : {SEND_REQUEST : this._topicNamespace + '/send/request'},
+    subList : {},
 
     /**
      * @constructor
@@ -62,7 +53,7 @@
             this.transport.addResponseFunctions(this.handleResponse, this.handleError);
 
             //listen out for other classes wanting to send requests to the server
-            topic.subscribe(this.subList.SEND_REQUEST, this, this.send);
+            topic.subscribe(Globals.SEND_REQUEST, this, this.send);
 
             //listen out for user level events
             topic.subscribe(Globals.TOPIC_USER_LOGOUT, this, this.handleLogout);
Index: src/js/rishson/enterprise/control/Transport.js
===================================================================
--- src/js/rishson/enterprise/control/Transport.js	(date 1320189221000)
+++ src/js/rishson/enterprise/control/Transport.js	(revision )
@@ -26,7 +26,7 @@
      * @description Mitigate CSRF attacks using Double Cookie Submission technique
      * Get the id of the current browser session from this domain's cookies
      */
-     _sessionId : cookie.cookie("JSESSIONID"),
+     _sessionId : cookie("JSESSIONID"),
 
     /**
      * @field
